app/models/office.rb

```diff
@@ -8,4 +8,49 @@ class Office < ApplicationRecord
   validates :region, numericality: { only_integer: true }
 
   accepts_nested_attributes_for :address, update_only: true
+
+  scope :with_claimed_shifts, -> { joins(needs: :shifts).merge(Shift.claimed).reorder(nil) }
+  scope :claimed_shifts_by_office, -> { with_claimed_shifts.group('offices.id') }
```

Comment:
personally, I'd use class methods for any of these that aren't chainable. 

http://blog.plataformatec.com.br/2013/02/active-record-scopes-vs-class-methods/

Old post, but think it has some good info

Reply:
Note the "Scopes are always chainable" section